app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,81 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, only: %i[edit update destroy]
```

Comment:
Not only. According to [that](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png), you've forgotten `new` and `create`

Reply:
we fix the issues